### PR TITLE
Change future value behavior of message state

### PR DESCRIPTION
**Pull request by rakshith91**

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py [hunk 1]
```diff
@@ -971,17 +971,19 @@ def dead_letter_source(self):
...
 
-        :rtype: ~azure.servicebus.ServiceBusMessageState
+        :rtype: int
         """
         try:
             message_state = self._raw_amqp_message.annotations.get(MESSAGE_STATE_NAME)
             if not message_state:
                 return ServiceBusMessageState.ACTIVE
+            elif message_state not in (0, 1, 2):
+                return message_state
```
Comment (annatisch) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py:986]: I think you could simply use:
```python
try:
    return ServiceBusMessageState(message_state)
except ValueError:
    return  message_state
```

**Review by yunhaoling — APPROVED** [review]
do we need to remove this `ServiceBusMessageState` class from our api?

**Comment by rakshith91** [comment]
> do we need to remove this ServiceBusMessageState class from our api?

we would still need it :)

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py (hunk 1 above)
Comment (rakshith91) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py:986]: it'll be more like 
```
            try:
                return ServiceBusMessageState(message_state)
            except ValueError:
                if not message_state:
                    return ServiceBusMessageState.ACTIVE
                return message_state
```

Comment (annatisch) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py]: Yep, though I was just referring to the `elif`.... so:
```python
if not message_state:
    return ServiceBusMessageState.ACTIVE
try:
    return ServiceBusMessageState(message_state)
except ValueError:
    return message_state
```
But either/or. both should work.

**Pull request merged**